src/common/com/intellij/plugins/haxe/model/HaxeParameterModel.java

```diff
@@ -90,8 +95,16 @@ public HaxeTypeTag getTypeTagPsi() {
     return parameter.getTypeTag();
   }
 
-  public SpecificTypeReference getType() {
-    return HaxeTypeResolver.getTypeFromTypeTag(getTypeTagPsi());
+  public ResultHolder getType() {
+    return getType(null);
+  }
+
+  public ResultHolder getType(@Nullable HaxeGenericResolver resolver) {
+    if (resolver != null) {
+      ResultHolder resolved = resolver.resolve(getType(null).getType().toStringWithoutConstant());
```

Comment:
I will try to improve the way it reads this part.
This was the first time I did generic resolving and unknown mutation so I was experimenting and learning while coding, and it still requires a lot of polishing. It works, at least partially, but I will try to improve the code structure.